sympy/printing/tests/test_lambdarepr.py

```diff
@@ -177,5 +177,26 @@ def test_multiple_sums():
     assert (lambdify((x, a, b, c, d), s)(2, 3, 4, 5, 6) ==
             s.subs([(x, 2), (a, 3), (b, 4), (c, 5), (d, 6)]).doit())
 
+
 def test_settings():
     raises(TypeError, lambda: lambdarepr(sin(x), method="garbage"))
+
+
+def test_printmethod():
+    # In each case, printmethod is called to test
+    # its working
+
+    obj = LambdaPrinter()
+    assert obj.printmethod == "_lambda"
```

Comment:
I was rather thinking of a test that creates a custom object with each print method, and making sure it works correctly, like

```py
class Test(Expr):
    def _lambda(self, printer):
        return 'lambda'

    def _tensorflow(self, printer):
        return 'tensorflow'

    ...

obj = Test()
assert LambdaPrinter().doprint(obj) == 'lambda'
assert TensorflowPrinter().doprint(obj) == 'tensorflow'
...
```

That way we know they really work (I think thus far, printmethod is untested for printer subclasses; it's possible it doesn't work correctly, in which case the printer superclass will need to be fixed).